src/api/thegraph/TheGraphApi.ts

```diff
@@ -146,12 +148,18 @@ export class TheGraphApiImpl {
     }, {})
   }
 
-  private calculatePrice(
-    { priceInOwl, token: { decimals = DEFAULT_PRECISION } }: PriceEntry,
-    inWei: boolean,
-  ): BigNumber {
-    const price = new BigNumber(priceInOwl)
+  private calculatePrice(priceEntry: PriceEntry, inWei: boolean): BigNumber {
+    const {
+      priceInOwlNumerator,
+      priceInOwlDenominator,
+      token: { decimals = DEFAULT_PRECISION },
+    } = priceEntry
 
-    return inWei ? price : price.dividedBy(TEN_BIG_NUMBER.exponentiatedBy(decimals))
+    const numerator = new BigNumber(priceInOwlNumerator)
+    const denominator = new BigNumber(priceInOwlDenominator)
+
+    const price = numerator.dividedBy(denominator)
+
+    return inWei ? price.multipliedBy(TEN_BIG_NUMBER.exponentiatedBy(decimals)) : price
```

Comment:
Does `priceInOwlNumerator/priceInOwlDenominator` already account for decimal?

Reply:
Yes it does now in the latest version!